packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -334,7 +338,7 @@ function transform(codeGen: CodeGen): t.Expression {
             ) {
                 return codeGen.genScopedFragId(expression);
             }
-            if (isSvgUseHref(elmName, attrName, namespace)) {
+            if (isSvgUseHref(elmName, attrName, namespace) && addLegacySanitizationHook) {
```

Comment:
Flip the `addLegacySanitizationHook` condition since the boolean check will be faster to execute than invoking `isSvgUseHref`.

```suggestion
            if (addLegacySanitizationHook && isSvgUseHref(elmName, attrName, namespace)) {
```